lib/plugins/authldap/auth.php

```diff
@@ -463,9 +463,12 @@ protected function _constructPattern($filter) {
      * @return string
      */
     protected function _filterEscape($string) {
-        return preg_replace(
-            '/([\x00-\x1F\*\(\)\\\\])/e',
-            '"\\\\\".join("",unpack("H2","$1"))',
+        // see https://github.com/adldap/adLDAP/issues/22
+        return preg_replace_callback(
+            '/([\x00-\x1F\*\(\)\\\\])/',
+            function ($matches) {
+                return "\\".join("", unpack("H2", $matches[1]));
```

Comment:
Okay makes sense. I guess ultimately we should have some tests for this function.